scripts/build-docs.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-env node */
+'use strict';
+
+const path = require('path');
+const fsThen = require('fs-then-native');
+const mkdirp = require('mkdirp');
+
+const DEST_DIR = 'dist';
+const GITHUB_FORMAT = 'github';
+const KRAMDOWN_FORMAT = 'kramdown';
+const PC_DEV_CENTER_FORMAT = 'purecloudDevCenter';
+const SUPPORTED_DOC_OUTPUT_FORMATS = [GITHUB_FORMAT, KRAMDOWN_FORMAT, PC_DEV_CENTER_FORMAT];
+let docMdOutputFormat = process.env.DOC_MD_OUTPUT_FORMAT || PC_DEV_CENTER_FORMAT;
+if (SUPPORTED_DOC_OUTPUT_FORMATS.indexOf(docMdOutputFormat) < 0) {
+    console.error(`Unknown MD Output Format Specified: '${docMdOutputFormat}'`);
+    process.exit(1);
+}
+
+let relativeLinkExtension = '.md';
+if (docMdOutputFormat === PC_DEV_CENTER_FORMAT) {
+    relativeLinkExtension = '.html';
+}
+
+let flagDocsAsPreview = true;
+if (process.env.FLAG_DOCS_AS_PREVIEW) {
+    ['false', 'f', '0'].forEach(currFalseValue => {
+        if (process.env.FLAG_DOCS_AS_PREVIEW === currFalseValue) {
+            flagDocsAsPreview = false;
+        }
+    });
+}
+
+// If invoked via "npm run docs" or "node scripts/build-docs.js" run immediately
+const calledFromCli = !module.parent;
+if (calledFromCli) {
+    buildDocs();
```

Comment:
Thoughts on adding a catch handler here, log to the console, and then process.exit(1).  This will decouple the impl of buildDocs from it's invoking context.